module/VuFind/src/VuFind/ILS/Driver/KohaRest.php

```diff
@@ -2449,22 +2470,37 @@ protected function mapRenewalBlockReason($reason, $itype)
     }
 
     /**
-     * Return a location for a Koha item
+     * Return a location (branch or shelving) for a Koha item
      *
      * @param array $item Item
      *
      * @return string
      */
     protected function getItemLocationName($item)
     {
-        $libraryId = (!$this->useHomeLibrary && null !== $item['holding_library_id'])
-            ? $item['holding_library_id'] : $item['home_library_id'];
-        $name = $this->translateLocation($libraryId);
-        if ($name === $libraryId) {
-            $libraries = $this->getLibraries();
-            $name = isset($libraries[$libraryId])
-                ? $libraries[$libraryId]['name'] : $libraryId;
+        $name = '';
+
+        switch ($this->config['ItemTypeRenewalBlockMappings']['Location']) {
```

Comment:
I still think you should create a default value for the configuration in case it is omitted from the config file. It might also be worth doing some simple normalization to be more tolerant of slight variations. For example, you could do something like:

```
$locationConfig = strtolower(trim($this->config['ItemTypeRenewalBlockMappings']['Location'] ?? 'branch'));
switch ($locationConfig) {
    case 'shelving':
```

That prevents whitespace and case sensitivity problems in addition to providing the legacy behavior as a default. What do you think?

Reply:
Maybe it would be better to normalize the config setting in the init method? It might cause it to be done once even if not needed, but would avoid repeating the same code for each item.

Reply:
That makes sense to me. Adding a new property that gets initialized in the init() method might improve readability elsewhere. What do you think, @alexklbuckley?

Reply:
I agree! I've added that in my latest follow-up. thanks both!